big-practice/src/js/dashboard.js

```diff
@@ -36,11 +57,67 @@ const renderTable = () => {
 
       tableBody.innerHTML = tableTemplate;
     })
+
+    // Display error message when retrieving data from Json server
     .catch((error) => alert('An error occurred while getting movie', error));
 };
 
+/**
+ * Create new movie and save to database
+ */
+const createMovie = (data) => {
+  const option = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(data),
+  };
+
+  fetch(moviesApi, option)
+    // Parses JSON response into native JavaScript objects
+    .then((response) => response.json())
+    .then(() => renderTable())
+    // Show error message when API call is wrong
+    .catch((error) => alert('An error occurred while creating movie', error));
+};
+
+const handleCreateForm = () => {
+  const name = nameMovieInput.value;
+  const director = directorInput.value;
+  const nation = nationInput.value;
+  const url = `${moviesApi}?name=${name}`;
+
+  if (!isValidForm()) {
+    return;
+  }
+
+  fetch(url, { method: 'GET' })
+    .then((response) => response.json())
+    .then((nameMovie) => {
+      if (nameMovie.length === 0) {
+        const formData = { name, director, nation };
+
+        createMovie(formData);
+        modal.classList.remove('modal-show');
```

Comment:
Instead of using this way, could you create function on show/hide modal. Ex
hide(modal);
show(model);